lib/hawkular/alerts/alerts_api.rb

```diff
@@ -81,12 +81,100 @@ def create_trigger(trigger, conditions = [], dampenings = [], _actions = [])
       http_post 'triggers/trigger', full_trigger
     end
 
+    # Creates the group trigger definition.
+    # @param [Trigger] trigger The group trigger to be created
+    # @return [Trigger] The newly created group trigger
+    def create_group_trigger(trigger)
+      ret = http_post 'triggers/groups', trigger.to_h
+      Trigger.new(ret)
+    end
+
+    # Updates a given group trigger definition
+    # @param [String] trigger_id ID of the group trigger to update
+    # @param [Trigger] trigger the group trigger to be updated
+    # @return [Trigger] The updated group trigger
+    def update_group_trigger(trigger_id, trigger)
+      http_put "triggers/groups/#{trigger_id}/", trigger.to_h
+      get_single_trigger trigger_id, false
+    end
+
+    # Creates the group conditions definitions.
+    # @param [String] trigger_id ID of the group trigger to set conditions
+    # @param [String] trigger_mode Mode of the trigger where conditions are attached (:FIRING, :AUTORESOLVE)
+    # @param [GroupConditionsInfo] group_conditions_info the conditions to set into the group trigger with the mapping
+    #                                                    with the data_id members map
```

Comment:
Perhaps useful to add the notes: 
* The dataIdMemberMap should be null if the group has no members. 
* The dataIdMemberMap should be null if this is a DataDriven group trigger. In this case the member triggers are removed and will be re-populated as incoming data demands. 
* For [non-data-driven] group triggers with existing members the dataIdMemberMap is handled as follows. For members not included in the dataIdMemberMap their most recently supplied dataIdMap will be used. This means that it is not necessary to supply mappings if the new condition set uses only dataIds found in the old condition set. If the new conditions introduce new dataIds a full dataIdMemberMap must be supplied